app/validators/previously_published_validator.rb

```diff
@@ -0,0 +1,14 @@
+class PreviouslyPublishedValidator < ActiveModel::Validator
+  def validate(record)
+    if record.previously_published.nil?
+      record.errors[:base] << "You must specify whether the document has been published before"
+      record.has_previously_published_error = true
```

Comment:
Do you want an inverse of this anywhere? I assume once you have this set to true and change state of the model there is nothing to set this back to false? (not sure that matters)